### PR TITLE
Fix some markdown issues

### PR DESCRIPTION
**Pull request by iamalicecarroll**
There are several things yet to be considered:

- [ ] Every file in [`docs/`](https://github.com/kdr-aus/ogma/tree/main/docs) starts with 

   ```html
   <iframe src="./.ibox.html?raw=true" style="border:none; position:fixed; width:40px; right:0; z-index=999;"></iframe>
   ```

   MDL doesn't like that because:

   - It's inline HTML
   - The first line should be a top-level heading
	
   Maybe the documentation renderer could add that?
- [ ] [`docs/15 Examples/15.3 Calculating Pi.md`](https://github.com/iamalicecarroll/ogma/blob/fix-markdown/docs/15%20Examples/15.3%20Calculating%20Pi.md?plain=1) [has](https://github.com/iamalicecarroll/ogma/blob/4af974ba35885d4e4a505ca69413f70c78de49e8/docs/15%20Examples/15.3%20Calculating%20Pi.md?plain=1#L22-L24) an embedded YouTube video:
   
   ```html
   <iframe title="YouTube video player" src="https://www.youtube.com/embed/sFun7N9OpiQ"
   width="560" height="315" frameborder="0" allowfullscreen
   allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"></iframe>
   ```

   which MDL doesn't like as it's inline HTML
   Daedalus seems to already support embedded local videos, maybe it could also have YouTube/Vimeo/etc. support? Although this is a Daedalus issue.
- [ ] Every second-level heading (`## Something`) is followed by a `---`, for example [`docs/01 Introduction.md`](https://github.com/iamalicecarroll/ogma/blob/fix-markdown/docs/01%20Introduction.md?plain=1) ([permalink](https://github.com/iamalicecarroll/ogma/blob/4af974ba35885d4e4a505ca69413f70c78de49e8/docs/01%20Introduction.md?plain=1#L12-L14)):

   ```md
   ## Characteristics
   
   ---
   ```

   Maybe this could be done by easy CSS like in GitHub? A bit modified example: 

   ```css
   h2 {
       padding-bottom: .3em;
       border-bottom: 1px solid gray;
   }
   ```

   But there's one [exception](https://github.com/iamalicecarroll/ogma/blob/4af974ba35885d4e4a505ca69413f70c78de49e8/docs/05%20Syntax%20and%20Semantics/5.1%20Syntax.md?plain=1#L69-L71) which confuses me in [`docs/05 Syntax and Semantics/5.1 Syntax.md`](https://github.com/iamalicecarroll/ogma/blob/fix-markdown/docs/05%20Syntax%20and%20Semantics/5.1%20Syntax.md?plain=1): 

   ```md
   ## Sub-expressions without the braces
   
   There exists a shorthand for sub-expressions which can be used as the **_last_** argument. The  
   ```

   It's the only place where `h2` isn't followed by an `hr`.

---

could this also be `hacktoberfest-accepted`?

### CODE REVIEW

**Comment by kurtlawrence** [comment]
Thanks for the PR @iamalicecarroll.

Are you by chance running windows? If so you can run a local daedalus server and view the rendered docs.
I suspect it might look somewhat different with `ogma` codeblock language specifications instead of `plaintext` :stuck_out_tongue: 

That's okay if not, I'll go through this PR when I get a chance and just make sure it renders correctly in daedalus.

**Comment by iamalicecarroll** [comment]
Oops, I messed up somewhere and got a0c7616 and 3655d7b here

**Comment by iamalicecarroll** [comment]
> Thanks for the PR @iamalicecarroll.
> 
> Are you by chance running windows? If so you can run a local daedalus server and view the rendered docs. I suspect it might look somewhat different with `ogma` codeblock language specifications instead of `plaintext` stuck_out_tongue
> 
> That's okay if not, I'll go through this PR when I get a chance and just make sure it renders correctly in daedalus.

No, I use Arch Linux

**Comment by iamalicecarroll** [comment]
> Thanks for the PR @iamalicecarroll.
> 
> Are you by chance running windows? If so you can run a local daedalus server and view the rendered docs. I suspect it might look somewhat different with `ogma` codeblock language specifications instead of `plaintext` stuck_out_tongue
> 
> That's okay if not, I'll go through this PR when I get a chance and just make sure it renders correctly in daedalus.

I managed to run daedalus in WINE, I don't quite understand how do I open the ogma docs

**Comment by kurtlawrence** [comment]
Daedalus runs a local server, so you'd need to run it over a folder containing the docs. I'll look into releasing prebuilt Linux binaries, but that is probably a while away.

**Comment by iamalicecarroll** [comment]
> I'll look into releasing prebuilt Linux binaries, but that is probably a while away.

Could I help with that?

**Comment by kurtlawrence** [comment]
Unfortunately not, daedalus is currently closed source =/

**Comment by iamalicecarroll** [comment]
> Thanks for the PR @iamalicecarroll.
> 
> Are you by chance running windows? If so you can run a local daedalus server and view the rendered docs. I suspect it might look somewhat different with `ogma` codeblock language specifications instead of `plaintext` 😛
> 
> That's okay if not, I'll go through this PR when I get a chance and just make sure it renders correctly in daedalus.

@kurtlawrence sorry for disturbing, have you checked it yet?

**Comment by iamalicecarroll** [comment]
Also what do you think about the tasks in the PR description?

**Comment by kurtlawrence** [comment]
Hi @iamalicecarroll, I haven't found the time to check this yet, unfortunately. With my current schedule, I might not get time until the new year =/
I do like the tasks =)

File: README.md [hunk 1]
```diff
@@ -10,26 +10,26 @@ efficiently work with tabular data.
 
 ![](./docs/assets/common-cmds.filter.gif?raw=true)
 
-# Getting Started
+## Getting Started
```
Comment (kurtlawrence) [inline README.md:13]: I'd like to keep these as H1 headings.

File: docs/01 Introduction.md [hunk 2]
```diff
@@ -10,50 +10,57 @@ The language is syntactically lightweight yet boasts powerful constructs designe
...
+
   - The ogma evaluation process sits between compiled and interpreted. When an expression is processed,
-each atomic block of the expression is 'compiled' into an evaluation unit, with only knowledge of
-_types_. This series of evaluation 'steps' is then evaluated with actual _values_. The split
-between the 'typing' and the 'value' makes evaluations of repeated expressions (such as _mapping_
-rows in a column) very fast since the typing evaluation has already been done.
+    each atomic block of the expression is 'compiled' into an evaluation unit, with only knowledge of
+    _types_. This series of evaluation 'steps' is then evaluated with actual _values_. The split
+    between the 'typing' and the 'value' makes evaluations of repeated expressions (such as _mapping_
+    rows in a column) very fast since the typing evaluation has already been done.
```
Comment (kurtlawrence) [inline docs/01 Introduction.md:48]: Much of this formatting does not impact the rendering.

File: docs/02 Getting Started/2.1 Installation.md [hunk 3]
```diff
@@ -12,46 +14,53 @@ _It is recommended to watch the ogma repository for notification of the latest r
 ### Linux
 
 1. Fetch the latest Linux binaries:
-```sh
-curl -L -o ogma.tar.gz https://github.com/kdr-aus/ogma/releases/latest/download/linux.tar.gz
-```
+
+   ```sh
+   curl -L -o ogma.tar.gz https://github.com/kdr-aus/ogma/releases/latest/download/linux.tar.gz
+   ```
```
Comment (kurtlawrence) [inline docs/02 Getting Started/2.1 Installation.md:20]: I would not indent this.

**Review by kurtlawrence — COMMENTED** [review]
Hi @iamalicecarroll,

I have reviewed the changes and found that they are insubstantial.

The changes are mostly aesthetic and do not impact the rendering to HTML. I do not think it is necessary to enforce a markdown style when the _rendering_ is the goal.

Thank you for submitting a PR, unfortunately I will not be moving forward with it.

**Pull request closed without merging**